### DISCUSSION TITLE
selector next (up) and previous (down)  PC

### DISCUSSION THREAD

**Discussion by santta** [tttapa/Control-Surface#473] [Ideas]
Hello, does the library have selectors to switch PC up and down?
One button switches up the PC, the second button lowers the PC.

**Comment by tttapa** [comment]
If you have a handful of programs, you can use the [Program-Changer](https://tttapa.github.io/Control-Surface-doc/Doxygen/d9/d7f/Program-Changer_8ino-example.html) or [ManyAddressesPCButton](https://tttapa.github.io/Control-Surface-doc/Doxygen/dc/db4/ManyAddressesPCButton_8ino-example.html) examples with an [IncrementDecrementSelector](https://tttapa.github.io/Control-Surface-doc/Doxygen/d2/deb/classIncrementDecrementSelector.html#details).

If you want a contiguous range of program numbers, you could try something like this:
```cpp
#include <Control_Surface.h>

template <setting_t NumPrograms = 128>
class FullRangeProgramChanger : public Selectable<NumPrograms> {
 public:
   FullRangeProgramChanger(MIDIAddress firstProgram = 0) : firstProgram(firstProgram) {}

   void select(setting_t setting) override {
       Control_Surface.sendPC(firstProgram + this->validateSetting(setting));
   }

 private:
   MIDIAddress firstProgram;
};

USBMIDI_Interface midi;

FullRangeProgramChanger<128> pc; // can select any of the 128 MIDI programs
IncrementDecrementSelector<128> selector {
    pc,         // this selector controls the program changer
    {2, 3},     // push button pins (increment, decrement)
    Wrap::Wrap, // Wrap around
};

void setup() {
  Control_Surface.begin();
}

void loop() {
  Control_Surface.loop();
}
```

**Reply by santta** [reply]
thanks, I will try this sketch, and as always I will share the finished code)))
